back/engines/commercial/insights/app/services/insights/category_suggestions_service.rb

```diff
@@ -0,0 +1,104 @@
+# frozen_string_literal: true
+
+module Insights
+  # This service provides utils to interact with the nlp service for category 
+  # suggestions.
+  #
+  # It can be used to trigger requests to the nlp service for category
+  # classification and to store prediction results.
+  class CategorySuggestionsService
+    class << self
+      # @param [NLP::ZeroshotClassificationResult] zsc_result zeroshot-classification result
+      # @return [Array<Insights::CategoryAssignment>]
+      def save_suggestion(zsc_result)
+        return unless zsc_result.success?
+
+        Tenant.find(zsc_result.tenant_id).switch do
+          zsc_task = ZeroshotClassificationTask.find_by(task_id: zsc_result.task_id)
+          return [] unless zsc_task
+
+          zsc_task.destroy!
+          save_predictions(zsc_result.predictions)
```

Comment:
Would be nice to also describe what we expect the shape of predictions to look like

Reply:
Thanks for the review! Not sure to understand what you meant here. The idea behind the Prediction class is that you don't really have to know the JSON structure of the response. The only place where the JSON structure still matters is in the Prediction class where the response is parsed. (I added an extra comment there.)